src/invidious/errors/special_exceptions.cr

```diff
@@ -0,0 +1,59 @@
+#
+# This file contains special exceptions whose error page differ from the norm.
+#
+
+# InfoExceptions are for displaying information to the user.
+#
+# An InfoException might or might not indicate that something went wrong.
+# Historically Invidious didn't differentiate between these two options, so to
+# maintain previous functionality InfoExceptions do not print backtraces.
+class InfoException < Exception
+end
+
+# InitialInnerTubeParseExceptions are for used to display extra information on
+# the error page for debugging/research purposes.
+#
+class InitialInnerTubeParseException < Exception
+  def initialize(@endpoint : String, @client_config : String, @data : String, @status_code : Int32, @mime_type : String, @cause : Exception)
+  end
+
+  def self.new(parse_exception : Exception,
```

Comment:
Why'd you make this new method? If it's necessary, could you call this new method generated by the initialize instead of doing the allocation and calling initialize yourself?